src/apps/queues/views.py

```diff
@@ -0,0 +1,76 @@
+# from braces.views import LoginRequiredMixin
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.exceptions import PermissionDenied, ValidationError
+# from django.core.urlresolvers import reverse
+from django.db.models import Q
+from django.urls import reverse
+from django.forms.utils import ErrorList
+from django.http import HttpResponse
+from django.http import HttpResponseRedirect
+from django.views.generic import ListView, CreateView, UpdateView, DeleteView
+from pyrabbit.http import HTTPError
+
+from .forms import QueueForm
+from .models import Queue
+from . import rabbit
+
+
+class QueueFormMixin(object):
+    def get_success_url(self):
+        return reverse('queues:list')
+
+    def get_form_kwargs(self, *args, **kwargs):
+        kwargs = super(QueueFormMixin, self).get_form_kwargs(*args, **kwargs)
+        kwargs['user'] = self.request.user
+        return kwargs
+
+
+class QueueListView(LoginRequiredMixin, ListView):
+    model = Queue
+    template_name = 'queues/list.html'
+
+    def get_queryset(self):
+        qs = Queue.objects.filter(owner=self.request.user) | self.request.user.organized_queues.all()
+        return qs
+
+
+class QueueCreateView(LoginRequiredMixin, QueueFormMixin, CreateView):
```

Comment:
Do these all need to be separate views? Can we combine Create and Update (and maybe delete? though that seems unlikely) with a FormView?

Reply:
Hmm. I'm thinking we take just a quick look into using FormView, and if it looks like it'll be all-encompassing, than switch. But I suppose there is some benefit to the explicitness of having these all broken out. I think I'm just used to looking at ViewSets where these are abstracted away from us, so it seems weird to have it explicit.